gateway/api/user/user.go

```diff
@@ -81,13 +81,14 @@ func (u *Router) login(w http.ResponseWriter, r *http.Request) {
 }
 
 func (u *Router) verify(w http.ResponseWriter, r *http.Request) {
+	email := r.FormValue("email")
 	hash := r.FormValue("hash")
-	if hash == "" {
-		render.Render(w, r, res.ErrBadRequest(r, "hash is required"))
+	if hash == "" || email == "" {
+		render.Render(w, r, res.ErrBadRequest(r, "Email and hash are required"))
```

Comment:
Go errors, by convention, start with a lower case - we might want to stick to that convention for our API responses as well

```suggestion
		render.Render(w, r, res.ErrBadRequest(r, "email and hash are required"))
```